service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RepresentativesController.java

```diff
@@ -0,0 +1,100 @@
+
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+
+@Api
+@RestController
+@RequestMapping("/callback/manage-representatives")
+public class RepresentativesController {
+
+    private final ObjectMapper mapper;
+    private final RepresentativeService representativeService;
+    private final RespondentService respondentService;
+    private final OthersService othersService;
+
+    @Autowired
+    public RepresentativesController(ObjectMapper mapper,
+                                     RepresentativeService representativeService,
+                                     RespondentService respondentService,
+                                     OthersService othersService) {
+        this.mapper = mapper;
+        this.representativeService = representativeService;
+        this.respondentService = respondentService;
+        this.othersService = othersService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        caseDetails.getData().put("representatives", representativeService.getDefaultRepresentatives(caseData));
+        caseDetails.getData().put("respondents_label", getRespondentsLabel(caseData));
+        caseDetails.getData().put("others_label", getOthersLabel(caseData));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(
+        @RequestHeader(value = "authorization") String authorisation,
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        final List<String> validationErrors = representativeService.validateRepresentatives(caseData, authorisation);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .errors(validationErrors)
+            .build();
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleSubmitted(
+        @RequestHeader(value = "authorization") String authorisation,
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        representativeService.addRepresentatives(caseData, caseDetails.getId(), authorisation);
+
+        caseDetails.getData().put("representatives", caseData.getRepresentatives());
```

Comment:
as above, if you decide to change it

Reply:
same, Map.of does not support null values